website/docs/deployment.mdx

```diff
@@ -383,19 +383,36 @@ on:
     # Review gh actions docs if you want to further define triggers, paths, etc
     # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on
 
+    paths:
+    - website/**
+
+defaults:
+  run:
+    working-directory: ./website
+
 jobs:
   deploy:
     name: Deploy to GitHub Pages
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        
+      - name: Set up Node
+        uses: actions/setup-node@v2
         with:
           node-version: 16.x
-          cache: yarn
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            **/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
```

Comment:
Please don't make these changes unless you know clearly why it doesn't work before and why it works after 😅 We assume users are using Yarn, not NPM. `package-lock.json` is NPM-only. We have explicitly stated that

> These files assume you are using yarn. If you use npm, change `cache: yarn`, `yarn install --frozen-lockfile`, `yarn build` to `cache: npm`, `npm ci`, `npm run build` accordingly.

Reply:
I used npm first(modifying all the commands of course), the error occurred. So, moved to yarn but that didn't solve the issue. I am not sure but [`cache-dependency-path`](https://github.com/actions/setup-node#caching-packages-dependencies) property might solve the issue. But, I didn't try that and instead used the code above cause, I didn't want to make the action files with the website and without the website look so different. The above code would work for both without specifying the path.

Reply:
We don't want to introduce another third-party action only for retrieving cache, especially one that gets even longer than the build/deploy step itself. As I said, the culprit is that you don't have a `yarn.lock` file at the root, which should be avoided at all costs. A well-setup repo should allow contributors to run `yarn install` at the root without `cd`ing into any subdirectory.

Reply:
I should not have used the npm in the first place. 😅

Reply:
The problem is not about the package manager, but the absence of a lock file at the root.

Reply:
But with `website/` structure would that be possible?

Reply:
Yes, you need a monorepo management tool. We (and the `react-svg-worldmap` repo I linked below) use Yarn workspaces. We have some documentation here: https://docusaurus.io/docs/next/installation#monorepos

Reply:
Got it.

Reply:
What if I am documenting a non-JS project? For example, the root contains a python project but I am documenting this with `website/` docusaurus.

Reply:
Maybe I should have mentioned this at the beginning😅. My root does not have JS codes. For me _run setup-node in the website folder_ does not work also. That's why I proposed `actions/cache@v2` as it would cover cases like mine. But it is understandable not wanting to add another third-party tool. Thanks for the prompt replies.